internal/resources/credential/datasource_credential.go

```diff
@@ -39,14 +44,59 @@ func dataSourceCredentialRead(ctx context.Context, d *schema.ResourceData, m int
 
 	name, _ := d.Get(NameKey).(string)
 
-	resp, err = config.TMCConnection.CredentialResourceService.CredentialResourceServiceGet(constructFullname(d))
-	if err != nil || resp == nil {
-		if clienterrors.IsNotFoundError(err) && !helper.IsDataRead(ctx) {
-			_ = schema.RemoveFromState(d, m)
-			return diags
+	getCredentialResourceRetrybleFn := func() (retry bool, err error) {
+		resp, err = config.TMCConnection.CredentialResourceService.CredentialResourceServiceGet(constructFullname(d))
+		if err != nil || resp == nil {
+			if clienterrors.IsNotFoundError(err) && !helper.IsDataRead(ctx) {
+				_ = schema.RemoveFromState(d, m)
+				return false, nil
+			}
+
+			return true, errors.Wrapf(err, "unable to get Tanzu Mission Control credential entry, name : %s", name)
 		}
 
-		return diag.FromErr(errors.Wrapf(err, "unable to get Tanzu Mission Control credential entry, name : %s", name))
+		if resp.Credential.Status == nil || resp.Credential.Status.Phase == nil {
+			return true, errors.Errorf("status or phase not found for Tanzu Mission Control credential entry, name: %s", name)
+		}
+
+		if !strings.EqualFold(string(*resp.Credential.Status.Phase), string(credentialsmodels.VmwareTanzuManageV1alpha1AccountCredentialStatusPhaseVALID)) {
+			log.Printf("[DEBUG] waiting for creadential(%s) to be in VALID phase", name)
+			return true, nil
+		}
+
+		return false, nil
+	}
+
+	timeoutData := d.Get(waitKey).(string)
+	if !helper.IsCreateState(ctx) {
+		timeoutData = helper.DoNotRetry
+	}
+
+	var timeoutDuration time.Duration
+
+	switch timeoutData {
+	case helper.DoNotRetry:
+		_, err = getCredentialResourceRetrybleFn()
+	case "default":
+		// nolint: ineffassign
```

Comment:
Why is the linter disabled?

Reply:
The reason is we assign some value to `timeoutDuration` in L82, which I actually overridden in L86/L92. I can in fact remove this case completely but I have just kept it for consistency with other resources. 

Assume we reach L82 then the value of `timeoutDuration` is overridden in L92 as L86 will hit parsing error due to `timeoutData = default`

More details about this linter can be found [here](https://pkg.go.dev/github.com/gordonklaus/ineffassign#section-readme)

Reply:
Let's remove the code that is added for consistency reason. It is always good to have better and cleaner code than following consistency.

Reply:
Only specifying the default timeout as `default` is added here for consistency.  Which code are you asking to remove?

Reply:
I have modified it to have the default timeout now.

Reply:
Thank you for addressing